### PR TITLE
Add support for branches in Client

### PR DESCRIPTION
**Pull request by convoliution**

### CODE REVIEW

File: client/verta/verta/_repository/commit.py [hunk 1]
```diff
@@ -15,13 +15,14 @@
...
 class Commit(object):
-    def __init__(self, conn, repo_id, parent_ids=None, id_=None):
+    def __init__(self, conn, repo_id, parent_ids=None, id_=None, branch_name=None):
         self._conn = conn
 
         self._repo_id = repo_id
         self._parent_ids = list(collections.OrderedDict.fromkeys(parent_ids or []))  # remove duplicates while maintaining order
 
         self.id = id_
+        self.branch_name = branch_name  # TODO: find a way to clear if branch is moved
```
Comment (conradoverta) [inline client/verta/verta/_repository/commit.py]: What do you mean moved? It thinks it has the branch top until we re-fetch from the remote, which gives the latest commit. I'm not sure what this to-do means, but not an issue.

File: client/verta/tests/test_repository.py [hunk 2]
```diff
@@ -23,17 +23,15 @@ def test_get_by_id(self, client, repository):
 
 class TestCommit:
     def test_add_get_rm(self, commit):
-        pytest.importorskip("boto3")
-
-        dataset1 = verta.dataset.S3("s3://verta-starter/census-test.csv")
-        dataset2 = verta.dataset.S3("s3://verta-starter/census-train.csv")
+        blob1 = verta.environment.Python(["a==1"])
```
Comment (conradoverta) [inline client/verta/tests/test_repository.py]: Please avoiding unrelated changes to make it easier to review.

File: client/verta/tests/test_repository.py [hunk 3]
```diff
@@ -80,69 +76,127 @@ def test_walk(self, commit):
...
+        path = "path/to/bananas"
+
+        commit = repository.new_commit()
+        commit.update(path, blob)
+        commit.save()
+        try:
+            commit.branch(branch)
+            assert repository.get_commit(branch=branch).id == commit.id
+        finally:
+            utils.delete_commit(repository.id, commit.id, repository._conn)
```
Comment (conradoverta) [inline client/verta/tests/test_repository.py]: What happens to the branch on this case? We need to be smart about it (backend should block maybe? Cc @ravi). Please make a ticket for us to think about it (same for tags) and cc me. Not a blocker.

File: client/verta/verta/_repository/commit.py (hunk 1 above)
Comment (convoliution) [reply client/verta/verta/_repository/commit.py]: This shouldn't come up in normal use, but if a user is juggling multiple `Commit` objects at once, it could be the case where one `Commit` used to be the tip of a branch, but isn't anymore, but that object still has the `branch_name` attr set.

File: client/verta/tests/test_repository.py (hunk 3 above)
Comment (convoliution) [reply client/verta/tests/test_repository.py]: Ticketed as VR-3554 & VR-3555.

**Comment by ravi** [comment]
Ew, Python. :-) Also, I am guessing you folks want some other Ravi?

**Comment by conradoverta** [comment]
Ha! Sorry about that. For some reason it didn't auto complete and I didn't notice. I meant @ravishetye. Thanks for being a good spirit =)
